added search_jobs function

kais-viz opened the pull request.

kais-viz commented:
This part adds job search function that returns a json object.
Also added it to the docs

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -102,7 +103,6 @@ def search(self, params, limit=None, results=[]):
             new_elements.extend(data["data"]["elements"][i]["elements"])
             # not entirely sure what extendedElements generally refers to - keyword search gives back a single job?
             # new_elements.extend(data["data"]["elements"][i]["extendedElements"])
-
```
are these from running black?

kais-viz replied:
Yes, I ran it on black playground online. I can rerun it and change some options so it doesn't omit the whitespaces you have in your code

tomquirk replied:
All good, no worries

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -213,9 +208,57 @@ def search_companies(self, keywords=None, limit=None):
+            f"/search/hits?{str_params}",
+            headers={
+                "accept": "application/vnd.linkedin.normalized+json+2.1",
+                "x-li-track": '{"clientVersion":"1.6.*","osName":"web","timezoneOffset":1,"deviceFormFactor":"DESKTOP","mpName":"voyager-web","displayDensity":1.100000023841858}',
```
**question:** is this essential?

kais-viz replied:
Yeah it is, I tried running it multiple times without the updated `x-li-track` but it returned the results without actually applying any of the filters apart from the query string and location. In order to add the additional filters it needed the new client version

tomquirk replied:
Very interesting, thanks for checking.  In this case, I'm happy to leave this here. I suspect we'll need to do something similar for the `application/vnd.linkedin.normalized+json+2.1` requests, but no worries for now.

tomquirk commented on tests/test_linkedin_api_requests.py:
```diff
@@ -149,6 +149,12 @@ def test_search_people_by_region(linkedin):
     assert results[0]["public_id"]
 
 
+def test_search_jobs(linkedin):
```
**praise**: for adding this test!

tomquirk commented on linkedin_api/linkedin.py:
```diff
@@ -213,9 +208,57 @@ def search_companies(self, keywords=None, limit=None):
-
         return results
 
+    def search_jobs(self, keywords, location, count=25, start=0, listedAt=86400):
```
**nitpick**: Can we change `listedAt` to `listed_at`?

kais-viz replied:
yeah ofcourse

kais-viz replied:
I reverted back the black changes to the file and only applied them to my part of the code, and changed listedAt to listed_at

kais-viz commented:
I reverted back the black changes to the file and only applied them to my part of the code, and changed listedAt to listed_at. 
Please have a look again!

tomquirk commented:
@kais-viz thanks for making those changes. LGTM! Thanks again for your contribution! 🎉 ✅ 🙌
